lib/withLoggedInUser.js

```diff
@@ -119,6 +132,12 @@ const withLoggedInUser = WrappedComponent => {
             // if we receive a 401 error from the API this means the 2FA token has failed
             throw new Error(newToken.message);
           }
+        } else if (recoveryCode) {
+          newToken = await verifyTwoFactorAuthenticationRecoveryCode(token, recoveryCode);
+          if (newToken.code === 401) {
```

Comment:
The return type for this function is a bit inconsistent, would be cleaner to throw this error rather than returning it